server/handlers.go

```diff
@@ -634,11 +634,27 @@ func (s *Server) handleToken(w http.ResponseWriter, r *http.Request) {
 		s.handleAuthCode(w, r, client)
 	case grantTypeRefreshToken:
 		s.handleRefreshToken(w, r, client)
+	case grantTypeClientCredentials:
+		s.handleClientCredentials(w, r, client)
```

Comment:
I think we need to copy code from my branch, because before entering the switch case we need to bind the user? At the moment the code uses the storage but this can't be reused.